src/Symfony/Component/Form/Extension/Core/Type/MoneyType.php

```diff
@@ -113,9 +113,9 @@ protected static function getPattern($currency)
             preg_match('/^([^\s\xc2\xa0]*)[\s\xc2\xa0]*123(?:[,.]0+)?[\s\xc2\xa0]*([^\s\xc2\xa0]*)$/u', $pattern, $matches);
 
             if (!empty($matches[1])) {
-                self::$patterns[$locale][$currency] = $matches[1].' {{ widget }}';
+                self::$patterns[$locale][$currency] = htmlentities($matches[1], ENT_QUOTES | ENT_SUBSTITUTE, 'UTF-8').' {{ widget }}';
```

Comment:
`protected getPattern` is called privately (self::getPattern) here, so no real BC issue i guess (assuming ppl might override moneytype returning a non-safe value or so).

We could move htmlentities to buildView perhaps.

Reply:
Does this part of the code know anything about HTML already?

Reply:
I guess not. Escaping from buildView() might be better.

Reply:
Or in the template directly (I don't know well this part of the code base)

Reply:
the issue is htmlentities vs. htmlspecialchars. We truly need htmlentites as escaping strategy, which is not possible in twig (AFAIK).

Reply:
with a formatter in the Twig bridge it becomes possible

Reply:
htmlentities formatter seems more something for core, dont you think. So from twig it would be like `{{ money_pattern | convert_encoding('UTF-8', _charset) | e('entities') }}` perhaps.

Reply:
Why not, you should try a PR on twig if you want to go down this path.
But without encoding conversion - entities cannot operate if the source encoding varies (or it needs to know about it, so that conversion should be handled there anyway)

Reply:
it will be a long road then; we cant use new twig features in 2.7 bridge i guess. 

Another path might be to inject kernel charset in moneytype, making it a service again. And go from there.

In general; it might become to much work. We could still settle with the current approach (keep a UTF-8 string; which happens today already).

Let me try a few things this week :)